strawberry/beta/pydantic/fields.py

```diff
@@ -0,0 +1,66 @@
+from decimal import Decimal
+from typing import Optional
+from uuid import UUID
+
+import pydantic
+
+from .exceptions import UnsupportedTypeError
+
+
+FIELDS_MAP = {
+    pydantic.NoneStr: Optional[str],
+    pydantic.NoneBytes: Optional[bytes],
+    pydantic.StrBytes: None,
+    pydantic.NoneStrBytes: None,
+    pydantic.StrictStr: str,
+    pydantic.ConstrainedBytes: bytes,
+    pydantic.conbytes: bytes,
+    pydantic.ConstrainedList: None,
+    pydantic.conlist: None,
+    pydantic.ConstrainedSet: None,
+    pydantic.conset: None,
+    pydantic.ConstrainedStr: str,
+    pydantic.constr: str,
+    pydantic.PyObject: None,
+    pydantic.ConstrainedInt: int,
+    pydantic.conint: int,
+    pydantic.PositiveInt: int,
+    pydantic.NegativeInt: int,
+    pydantic.ConstrainedFloat: float,
+    pydantic.confloat: float,
+    pydantic.PositiveFloat: float,
+    pydantic.NegativeFloat: float,
+    pydantic.ConstrainedDecimal: Decimal,
+    pydantic.condecimal: Decimal,
+    pydantic.UUID1: UUID,
+    pydantic.UUID3: UUID,
+    pydantic.UUID4: UUID,
+    pydantic.UUID5: UUID,
+    pydantic.FilePath: None,
+    pydantic.DirectoryPath: None,
+    pydantic.Json: None,
+    pydantic.JsonWrapper: None,
+    pydantic.SecretStr: str,
+    pydantic.SecretBytes: bytes,
+    pydantic.StrictBool: bool,
+    pydantic.StrictInt: int,
+    pydantic.StrictFloat: float,
+    pydantic.PaymentCardNumber: None,
+    pydantic.ByteSize: None,
+}
+
+
+def get_basic_type(type_):
+    if isinstance(type_, type):
+        if issubclass(type_, pydantic.ConstrainedInt):
+            return int
+        if issubclass(type_, pydantic.ConstrainedStr):
+            return str
+
+    if type_ in FIELDS_MAP:
+        type_ = FIELDS_MAP.get(type_)
+
+        if type_ is None:
+            raise UnsupportedTypeError()
```

Comment:
Would be nice to include the type that is unsupported and ideally the field that its assigned to.

Reply:
I am not sure how we can go from the "value" of the type alias, to the name... @patrick91 ?

Reply:
I think you'll need to pass the field to this function so that you can extract the name from it.

Reply:
the problem is that we won't get `pydantic.NoneStr` but `Union[str, None]` which doesn't look great 🤔

Reply:
I'm not suggesting this, but we could inspect the frame and check :D

Reply:
Oh I see, thats annoying. Well at least the field on the Pydantic model could be included in the error message? An error message like `Unsupported type for field "{field_name}" on Pydantic model "{pydantic_model_name}"` would make finding the offending field much easier.